File: Shantanu_Mamgain/Sliding-Window-Leetcode/Number of Substrings Containing All Three Characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int hash[] = {-1, -1, -1};
        
        int count = 0;
        int i = 0;
        while(i < s.length()){
            hash[s.charAt(i)-'a']= i;
            if(hash[0] > -1 && hash[1] > -1 && hash[2] > -1){
                count += Math.min(hash[0],Math.min(hash[1],hash[2]))+1;
            }
            i++;
        }
        return count;
    }
}

File: Lakshay Garg/Leetcode/Sliding Window/Number of Substrings Containing All Three Characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int end=0,ans=0;
        int arr[3]={-1,-1,-1};
        while(end<s.size()){
            arr[s[end]-'a']=end;
            if(arr[0]>-1 && arr[1]>-1 && arr[2]>-1) ans+=1+min(arr[0],min(arr[1],arr[2]));
            end++;
        }  
        return ans;
    }
};
